### PR TITLE
chore: update build instructions

### PR DESCRIPTION
**Pull request by tearfur**
1. Updated Transmission release verison in README.
2. Use more generic build commands (e.g. `make` -> `cmake --build`)
3. Combined commands.
4. Updated syntax highlighting.

### CODE REVIEW

**Review by ckerr — APPROVED** [review]
LGTM.

@mikedld, not sure if you're interested in this but CC'ing you because you've reviewed README.md changes in the past

File: README.md [hunk 1]
```diff
@@ -31,15 +31,16 @@ For a more detailed description, and dependencies, visit [How to Build Transmiss
...
-    $ sudo make install
+```bash
+$ tar xf transmission-4.0.4.tar.xz
+$ cd transmission-4.0.4
+# Use -DCMAKE_BUILD_TYPE=RelWithDebInfo to build optimized binary with debug information. (preferred)
+# Use -DCMAKE_BUILD_TYPE=Release to build full optimized binary.
+$ cmake -B build -DCMAKE_BUILD_TYPE=RelWithDebInfo
+$ cd build
+$ cmake --build .
+$ sudo cmake --install .
```
Comment (Coeur) [inline README.md:42]: You're building twice.

Can you limit it to max two instructions?
```
cmake -B build -DCMAKE_BUILD_TYPE=RelWithDebInfo
cmake -B build -DCMAKE_BUILD_TYPE=RelWithDebInfo --install
```

Comment (ckerr) [reply README.md:42]: I prefer using Ninja; but for the general-purpose-build-instructions-for-everyone `README.md`, it's probably better to omit any steps that aren't absolutely required.

Comment (ckerr) [reply README.md:42]: commenting again for clarification, I think I missed an important part: @Coeur will ninja be a build requirement due to Swift?

I'm OK with the idea, just making sure I understand correctly

Comment (Coeur) [reply README.md:42]: For the macOS app builds, `-G Ninja` OR `-G Xcode` will be required for Swift (`make` won't work). But ninja won't be strictly required to build: the Xcode option makes it buildable from command line without NInja.

Comment (GaryElshaw) [reply README.md:42]: If the current build system continues to work, shouldn't the proposed changes be added to the readme rather than replacing?

Comment (tearfur) [reply README.md:42]: I agree that the `README.md` build instructions should be as general as possible. Let's keep it as-is until the Swift PRs come.

**Pull request merged**